test/resolve-ref.ts

```diff
@@ -1,64 +1,280 @@
+                { name: 'should-be-tag', oid: '7b84f676f2fbea2a3c6d83924fa63059c7bdfbe2' },
                 { name: 'test-annotated-tag', oid: '56ea7a808e35df13e76fee92725a65a373a9835c' },
-                { name: 'test-tag', oid: '2dbee47a8d4f8d39e1168fad951b703ee05614d6' }
+                {
+                    name: 'test-annotated-tag/packed',
+                    oid: '56ea7a808e35df13e76fee92725a65a373a9835c'
+                },
+                { name: 'test-tag', oid: '2dbee47a8d4f8d39e1168fad951b703ee05614d6' },
+                {
+                    name: 'with-slash/should-be-tag',
+                    oid: '2dbee47a8d4f8d39e1168fad951b703ee05614d6'
+                }
             ];
 
             assert.deepStrictEqual(actual, expected);
         });
     });
+
+    describe('listRemotes()', () => {
+        it('remotes', async () => {
+            const actual = await repo.listRemotes();
+            const expected = ['origin', 'with-slash'];
+
+            assert.deepStrictEqual(actual, expected);
+        });
+    });
+
+    describe('resolveRef()', () => {
+        it('oid into oid', async () => {
+            const oid = '7b84f676f2fbea2a3c6d83924fa63059c7bdfbe2';
+            const actual = await repo.resolveRef(oid);
+
+            assert.strictEqual(actual, oid);
+        });
+
+        describe('ref into oid', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.resolveRef(testCase.ref);
+                    const expected = testCase.oid;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+
+        describe('refs starting with "ref: " ', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, async () => {
+                    const actual = await repo.resolveRef('ref: ' + testCase.ref);
+                    const expected = testCase.oid;
+
+                    assert.strictEqual(actual, expected);
+                });
+            }
+        });
+
+        describe('throws not found error with correct, but broken refs', () => {
+            for (const testCase of refFixtures) {
+                it(testCase.ref, () => {
+                    assert.rejects(
+                        () => repo.resolveRef(testCase.ref + 'n'),
+                        /ref not found refs\/heads\/mainn/
+                    );
+                });
+            }
+        });
+
+        describe('throws not found error with wrong ref', () => {
+            for (const testCase of wrongRefFixtures) {
+                it(testCase.ref, () => {
+                    assert.rejects(
+                        () => repo.resolveRef(testCase.ref + 'n'),
+                        /ref not found refs\/heads\/mainn/
+                    );
+                });
+            }
+        });
+
+        describe('shouldBeTag ref for ambiguous refs priority', () => {
```

Comment:
A description should describe what's expected behaviour, not a purpose/meaning/etc of something. Please, describe what's expected here.